### PR TITLE
Fix #245

### LINKED ISSUE #245: `linkWithUnclearPurpose` should only look at links, not `<a>` without `href`.
Ok:
```
<a id="this-is-an-anchor"></a>
```

Not ok:
```
<a id="this-is-a-link" href="/"></a>
```

### PR DESCRIPTION
**Pull request by alice**
@ricksbrown PTAL?

### CODE REVIEW

File: test/audits/link-with-unclear-purpose.js [hunk 1]
```diff
@@ -132,3 +132,13 @@ test('a hidden link should not be run against the audit', function() {
       rule.run({scope: fixture}),
       { result: axs.constants.AuditResult.NA });
 });
+
+test('an anchor tag without href attribute is ignored', function() {
+    var fixture = document.getElementById('qunit-fixture');
+    var a = fixture.appendChild(document.createElement('a'));
```
Comment (ricksbrown) [inline test/audits/link-with-unclear-purpose.js]: totally nitpicking here but `var a` is never used.

File: test/audits/link-with-unclear-purpose.js [hunk 2]
```diff
@@ -132,3 +132,13 @@ test('a hidden link should not be run against the audit', function() {
...
+
+    var rule = axs.AuditRules.getRule('linkWithUnclearPurpose');
+    deepEqual(
+      rule.run({scope: fixture}),
```
Comment (ricksbrown) [inline test/audits/link-with-unclear-purpose.js]: style nit: space after `{` and before `}`

**Comment by ricksbrown** [comment]
@alice couple of trifling issues.
This is a nice little patch, good stuff, LGTM.

File: test/audits/link-with-unclear-purpose.js (hunk 1 above)
Comment (alice) [inline test/audits/link-with-unclear-purpose.js]: Done.

File: test/audits/link-with-unclear-purpose.js (hunk 2 above)
Comment (alice) [inline test/audits/link-with-unclear-purpose.js]: Would fix this but it's consistent with the rest of the file, so I don't think it's worth it.

**Pull request merged**